.github/workflows/full_ci.yml

```diff
@@ -31,12 +31,12 @@ jobs:
       contents: read # to fetch code (actions/checkout)
 
     name: macOS
-    runs-on: macos-13
+    runs-on: macos-14
     strategy:
       fail-fast: true
       matrix:
         target: [
-          x86_64-apple-darwin,
```

Comment:
We probably shouldn't lose x86 MacOS yet because it is still Tier 1, are you able to add it back?

After that I think this can be merged together with https://github.com/rust-lang/libc/pull/3794 (in https://github.com/rust-lang/libc/pull/3795)

Reply:
Note it is only for CI also because further macos images are exclusively arm64.

Reply:
We would just need separate jobs that run on `macos-13` and `macos-14`, [which is what `rustc` currently does](https://github.com/rust-lang/rust/blob/6696447f784a888446d13bb400a8d507a68331c9/src/ci/github-actions/jobs.yml#L19-L25).

Was the x86_64-apple CI broken? If not, it may be better to drop this part of the change and move it to a new PR, so this PR is only the things needed to get CI working again.

Reply:
we would need to add exceptions for x86_64 with lack of certain types updates/changes, api missing and so on (even though in reality it s just because of macos releases not architecture related at all).

Reply:
cc @Amanieu

Reply:
Sounds good, thanks for confirming.